wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Auditor.kt

```diff
@@ -0,0 +1,100 @@
+package com.squareup.wire.schema
+
+import com.squareup.wire.schema.internal.TypeMover
+
+/**
+ * Extend this class to audit Wire, receive metrics, and validate them, including all [schema handlers][SchemaHandler]
+ * involved in the Protobuf schema manipulation.
+ *
+ * The events' order is as follows:
+ *  - start
+ *  - loadSchemaStart
+ *  - loadSchemaEnd
+ *  - treeShakeStart
+ *  - treeShakeEnd
+ *  - moveTypesStart
+ *  - moveTypesEnd
+ *  - schemaHandlersStart // Looping over all handlers.
+ *    - schemaHandlerStart
+ *    - schemaHandlerEnd
+ *  - schemaHandlersEnd
+ *  - validate
+ *  - end
+ */
+abstract class Auditor {
```

Comment:
Should we maybe call it "interceptor"? It's a common pattern, particularly in Square OS libraries. Don't hate "auditor", just feel like using a common name for a common pattern will aid comprehension.

Reply:
"EventListener" is also used in three or four libraries. I don't think this is an "interceptor" because you're not given an opportunity to actually intercept anything, just observe.

Reply:
I didn't wanna pick `EventListener` because I wanna convey it doesn't only listen, they can make the work throw with the `validate()` method.

Reply:
Hmmm I see. There's "visitor" or "callback" as other common nouns for things which are similar in nature. Not sure what else. I think "auditor" only stands out because it's not a term that you come across frequently and have intuition about.

Reply:
I didn't know about Visitor but I like it.